spring-integration-ip/src/main/java/org/springframework/integration/ip/udp/UnicastSendingMessageHandler.java

```diff
@@ -276,8 +297,10 @@ public void startAckThread() {
 	}
 
 	protected void send(DatagramPacket packet) throws Exception {
-		DatagramSocket socket = this.getSocket();
-		packet.setSocketAddress(this.getDestinationAddress());
+		DatagramSocket socket = getSocket();
+		if (this.socketExpression == null) {
+			packet.setSocketAddress(getDestinationAddress());
```

Comment:
How about `destination-expression` as mutually exclusive to the `host/port` pair?

Looks like I misunderstood it from the beginning and `socket` is independent object here from the `destinationAddress` in the datagram.

So, my new position that we should reinstate `socketExpression` setters and introduce that new one for the `destination` to be populated on the datagram to send. For this purpose we really have access to the `requestMessage` to extract receiving `host/port`. 

It isn't required to evaluate that expression to the `SocketAddress` because we can do something like this for the `host:port` string: http://stackoverflow.com/questions/2345063/java-common-way-to-validate-and-convert-hostport-to-inetsocketaddress.
Having that the expression can be evaluated to the `URI` as well.